### ISSUE TITLE
PDO exception raised on new installation

### ISSUE THREAD

**Issue by isleofwighter** [bakoontz/WikkaWiki#1306]
Running PHP7 and MySQL80 version 5.7.21
I fill in the configuration page and click continue and get this displayed:
Testing Configuration
Testing DB connection settings... OK

Checking availability of default pages... OK
Installing Stuff
Setting up database for UTF-8... OK

Fatal error: Uncaught PDOException: SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '-wikki DEFAULT CHARACTER SET utf8 COLLATE utf8_general_ci' at line 1 in D:\My Web\htdocs\wikki\setup\inc\db_update_mysql.php:13 Stack trace: #0 D:\My Web\htdocs\wikki\setup\inc\db_update_mysql.php(13): PDO->exec() #1 D:\My Web\htdocs\wikki\setup\install.php(89): require('D:\\My Web\\htdoc...') #2 D:\My Web\htdocs\wikki\wikka.php(648): include('D:\\My Web\\htdoc...') #3 {main} thrown in D:\My Web\htdocs\wikki\setup\inc\db_update_mysql.php on line 13

The create code includes reference to a time column and I single quoted this as it may be a reserved word. Same result. That is the end of my coepence with mySQL.
Has anyone an answer?

**Comment by bakoontz** [comment]
Hi!  What version of WW are you trying to install?  Are you installing from a pre-configured installer?  Tarball/zipfile?  git clone?

I test with MySQL 5.7.21 and am not getting this error.  Are you creating a brand new database?

**Comment by isleofwighter** [comment]
HI

Thanks for replying

Wikka-1.4.1.zip

It is a new schema in an existing MYSQL installation. Installed on Windows 10 with existing web server

The zip file extracted. to folder wikki under htdocs.

 

It is obviously something simple but I am not a professional developer.

Mike

**Comment by bakoontz** [comment]
The syntax is valid for 5.7.21:

https://dev.mysql.com/doc/refman/5.7/en/charset-database.html

I've had problems in the past with using hyphens with database names in MySQL.  Can you try creating a new database and replacing the hyphen with an underscore please?

**Comment by bakoontz** [comment]
Confirmed that this is, in fact, the issue:

PDOException: SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near **'-141 DEFAULT CHARACTER SET utf8 COLLATE utf8_unicode_ci'** at line 1 in C:\wamp64\www\WikkaWiki\setup\inc\db_update_mysql.php on line 13

(I created a database called wikkawiki-141 to run this test.)

It turns out the hyphen is not one of the allowed characters for unquoted identifiers in MySQL:

https://dev.mysql.com/doc/refman/5.7/en/identifiers.html

I do not routinely quote MySQL identifiers, and take care to choose column names that do not require quotes.  I would suggest using an underscore instead in your database, table, and column names as these do not require any special handling.

At some point I will do a check during installation that allows only identifiers not requiring quotes.
